src/main/kotlin/org/rust/cargo/runconfig/run/CargoRunRunConfigurationProducer.kt

```diff
@@ -0,0 +1,44 @@
+package org.rust.cargo.runconfig.run;
+
+import com.intellij.execution.actions.ConfigurationContext
+import com.intellij.execution.actions.RunConfigurationProducer
+import com.intellij.openapi.util.Ref
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.RustFnItem
+
+class CargoRunRunConfigurationProducer
+: RunConfigurationProducer<CargoRunRunConfiguration>(CargoRunRunConfigurationType()) {
+    override fun isConfigurationFromContext(configuration: CargoRunRunConfiguration,
+                                            context: ConfigurationContext): Boolean {
+        return configuration.name == "Run " + context.project?.name
```

Comment:
Ah, and you don't actually need `?`: again, because of the missing annotations Kotlin does not know that `context.project` is not null, so it does not warn about unnecessary  `?` here.